tox.ini

```diff
@@ -93,6 +93,15 @@ commands =
     --profile=prospector \
     --die-on-tool-error {posargs}
 
+[testenv:pre-commit]
+description = run pre-commit to ensure the changes are valid against our lintings
+deps =
+   pre-commit
+commands =
+    # FIXME: use `github.event.pull_request.base.sha` in `--from-ref` because if
+    # the base branch is different, this won't work as expected
+    pre-commit run --from-ref master --to-ref HEAD
```

Comment:
Does this change the files as well? Or do we need to run black/pre-commit by ourselves?

Reply:
This does change the files locally, but it does require a commit first -- this doesn't use staged files.

Reply:
I commented in the common PR as well. It does work on staged files to me. I'm not sure why it doesn't for you or if you have it configured in a different way.